### ISSUE TITLE
python says it installed but it did not

### ISSUE THREAD

**Issue by nmccready** [moonrepo/proto#209]
### What version?

0.18.1

### Which command?

proto install python 3.8

### What happened?

```bash
$ proto install python 3.8
━━         Installing Python 3.8.18                                                                                                                                                                                                    python-build: definition not found: 3.8.18
[ INFO 2023-09-18 17:39:38] proto::commands::install:install  Python has been installed to ~/.proto/tools/python/3.8.18!

~
$ ls ~/.proto/tools/python/3.8.18

# ... nothing
```

### Operating system?

MacOS

### Architecture?

x64

**Comment by milesj** [comment]
@nmccready Is there more to the logs (`--log trace`)?

Do you have `python-build` installed. I think it may be hitting that edge case and not working properly.

**Comment by nmccready** [comment]
```bash
$ proto --log=trace install python 3.8
[DEBUG 2023-09-18 17:58:01] proto  Running proto v0.18.1
[TRACE 17:58:01] starbase::app  Running startup phase with 1 systems
[TRACE 17:58:01] starbase::app  Running analyze phase with 1 systems
[TRACE 17:58:01] starbase::app  Running execute phase with 2 systems
[DEBUG 17:58:01] proto_core::tool_loader:install  Traversing upwards to find a configured plugin  tool="python"
[DEBUG 17:58:01] proto_core::tools_config:install:load  Loading .prototools  file="/Users/SOME_USER/code/c1/acme/data-ingestion/.prototools"
[TRACE 17:58:01] starbase_utils::fs:install:load  Opening file  file="/Users/SOME_USER/code/c1/acme/data-ingestion/.prototools"
[TRACE 17:58:01] starbase_utils::fs_lock:install:load  Locking file  file="/Users/SOME_USER/code/c1/acme/data-ingestion/.prototools"
[TRACE 17:58:01] starbase_utils::fs_lock:install:load  Unlocking file  file="/Users/SOME_USER/code/c1/acme/data-ingestion/.prototools"
[DEBUG 17:58:01] proto_core::tools_config:install:load  Loading .prototools  file="/Users/SOME_USER/.prototools"
[TRACE 17:58:01] starbase_utils::fs:install:load  Opening file  file="/Users/SOME_USER/.prototools"
[TRACE 17:58:01] starbase_utils::fs_lock:install:load  Locking file  file="/Users/SOME_USER/.prototools"
[TRACE 17:58:01] starbase_utils::fs_lock:install:load  Unlocking file  file="/Users/SOME_USER/.prototools"
[TRACE 17:58:02] warpgate::loader:install  Creating plugin loader  cache_dir="/Users/SOME_USER/.proto/plugins"
[TRACE 17:58:02] warpgate::loader:install  Loading plugin python  plugin="python" locator="source:https://github.com/moonrepo/python-plugin/releases/latest/download/python_plugin.wasm"
[TRACE 17:58:02] starbase_utils::fs:install  Reading file metadata  file="/Users/SOME_USER/.proto/plugins/python-latest-a51a99c7c1406ac222cbea14a1ec1dbaece72448f9b2f682cdfb511b0a21857b.wasm"
[TRACE 17:58:02] warpgate::loader:install  Plugin already downloaded and cached  plugin="python" path="/Users/SOME_USER/.proto/plugins/python-latest-a51a99c7c1406ac222cbea14a1ec1dbaece72448f9b2f682cdfb511b0a21857b.wasm"
[DEBUG 17:58:02] proto_core::tool_loader:install  Loading WASM plugin  source="/Users/SOME_USER/.proto/plugins/python-latest-a51a99c7c1406ac222cbea14a1ec1dbaece72448f9b2f682cdfb511b0a21857b.wasm"
[DEBUG 17:58:02] proto_core::tool:install  Creating tool python and instantiating plugin
[DEBUG 17:58:02] proto_core::tool_manifest:install:load  Loading manifest.json  file="/Users/SOME_USER/.proto/tools/python/manifest.json"
[DEBUG 17:58:03] proto_core::tool:install  Created tool python and its WASM runtime
[TRACE 17:58:03] warpgate::plugin:install  Calling plugin function register_tool  plugin="python" input={"id":"python"}
[TRACE 17:58:03] warpgate::plugin:install  Called plugin function register_tool  plugin="python" output={"env_vars":[],"inventory":{"disable_progress_bars":false},"name":"Python","plugin_version":"0.0.1","type_of":"Language"}
[DEBUG 17:58:03] proto_core::tool:install  Resolving a semantic version or alias  tool="python" initial_version="~3.8"
[DEBUG 17:58:03] proto_core::tool:install  Loading available versions  tool="python"
[TRACE 17:58:03] warpgate::plugin:install  Calling plugin function load_versions  plugin="python" input={"context":{"env_vars":{},"tool_dir":{"path":"/home/.proto/tools/python/latest","virtual_prefix":"/home","real_prefix":"/Users/SOME_USER"},"version":"latest"},"initial":"~3.8"}
[TRACE 17:58:03] proto_wasm::exec_command:install  Executing command from plugin  command="git" args=["ls-remote", "--tags", "--sort", "version:refname", "https://github.com/python/cpython"] env_vars={}
[TRACE 17:58:03] proto_wasm::exec_command:install  Executed command from plugin  command="git" exit_code=0 stderr_len=0 stdout_len=48175
[TRACE 17:58:03] warpgate::plugin:install  Called plugin function load_versions  plugin="python" output={"latest":"3.11.5","aliases":{"latest":"3.11.5"},"versions":["2.0.0","2.1.0","2.2.0","2.3.0","2.4.0","2.5.0","2.6.0","2.7.0","3.0.0","3.1.0","3.2.0","3.3.0","3.4.0","3.5.0","3.6.0","3.7.0","0.9.8","0.9.9","1.0.1","1.0.2","1.1.0","1.1.1","1.2.0","1.2.0-b.1","1.2.0-b.2","1.2.0-b.3","1.2.0-b.4","1.3.0","1.3.0-b.1","1.4.0","1.4.0-b.1","1.4.0-b.2","1.4.0-b.3","1.5.0","1.5.1","1.5.2","1.5.2-a.1","1.5.2-a.2","1.5.2-b.1","1.5.2-b.2","1.5.2-c.1","1.5.0-a.1","1.5.0-a.2","1.5.0-a.3","1.5.0-a.4","1.5.0-b.1","1.5.0-b.2","1.6.0-a.1","1.6.0-a.2","2.0.0","2.0.1","2.0.1-c.1","2.0.0-b.1","2.0.0-b.2","2.0.0-c.1","2.1.0","2.1.1","2.1.1-c.1","2.1.2","2.1.2-c.1","2.1.3","2.1.0-a.1","2.1.0-a.2","2.1.0-b.1","2.1.0-b.2","2.1.0-c.1","2.1.0-c.2","2.2.0","2.2.1","2.2.1-c.1","2.2.1-c.2","2.2.2","2.2.2-b.1","2.2.3","2.2.3-c.1","2.2.0-a.3","2.3.1","2.3.2","2.3.2-c.1","2.3.3","2.3.3-c.1","2.3.4","2.3.4-c.1","2.3.5","2.3.5-c.1","2.3.6","2.3.6-c.1","2.3.7","2.3.7-c.1","2.3.0-c.1","2.3.0-c.2","2.4.0","2.4.1","2.4.1-c.1","2.4.1-c.2","2.4.2","2.4.2-c.1","2.4.3","2.4.3-c.1","2.4.4","2.4.4-c.1","2.4.5","2.4.5-c.1","2.4.6","2.4.6-c.1","2.4.0-a.1","2.4.0-a.2","2.4.0-a.3","2.4.0-b.1","2.4.0-b.2","2.4.0-c.1","2.5.0","2.5.1","2.5.1-c.1","2.5.2","2.5.2-c.1","2.5.3","2.5.3-c.1","2.5.4","2.5.5","2.5.5-c.1","2.5.5-c.2","2.5.6","2.5.6-c.1","2.5.0-a.0","2.5.0-a.1","2.5.0-a.2","2.5.0-b.1","2.5.0-b.2","2.5.0-b.3","2.5.0-c.1","2.5.0-c.2","2.6.0","2.6.1","2.6.2","2.6.2-c.1","2.6.3","2.6.3-rc.1","2.6.4","2.6.4-rc.1","2.6.4-rc.2","2.6.5","2.6.5-rc.1","2.6.5-rc.2","2.6.6","2.6.6-rc.1","2.6.6-rc.2","2.6.7","2.6.8","2.6.8-rc.1","2.6.8-rc.2","2.6.9","2.6.9-rc.1","2.6.0-a.1","2.6.0-a.2","2.6.0-a.3","2.6.0-b.1","2.6.0-b.2","2.6.0-b.3","2.6.0-rc.1","2.6.0-rc.2","2.7.0","2.7.1","2.7.1-rc.1","2.7.2","2.7.2-rc.1","2.7.3","2.7.3-rc.1","2.7.3-rc.2","2.7.4","2.7.4-rc.1","2.7.5","2.7.6","2.7.6-rc.1","2.7.7","2.7.7-rc.1","2.7.8","2.7.9","2.7.9-rc.1","2.7.10","2.7.10-rc.1","2.7.11","2.7.11-rc.1","2.7.12","2.7.12-rc.1","2.7.13","2.7.13-rc.1","2.7.14","2.7.14-rc.1","2.7.15","2.7.15-rc.1","2.7.16","2.7.16-rc.1","2.7.17","2.7.17-rc.1","2.7.18","2.7.18-rc.1","2.7.0-a.1","2.7.0-a.2","2.7.0-a.3","2.7.0-a.4","2.7.0-b.1","2.7.0-b.2","2.7.0-rc.1","2.7.0-rc.2","3.0.0","3.0.1","3.0.0-a.1","3.0.0-a.2","3.0.0-a.3","3.0.0-a.4","3.0.0-a.5","3.0.0-b.1","3.0.0-b.2","3.0.0-b.3","3.0.0-rc.1","3.0.0-rc.2","3.0.0-rc.3","3.1.0","3.1.1","3.1.1-rc.1","3.1.2","3.1.2-rc.1","3.1.3","3.1.3-rc.1","3.1.4","3.1.4-rc.1","3.1.5","3.1.5-rc.1","3.1.5-rc.2","3.1.0-a.1","3.1.0-a.2","3.1.0-b.1","3.1.0-rc.1","3.1.0-rc.2","3.2.0","3.2.1","3.2.1-b.1","3.2.1-rc.1","3.2.1-rc.2","3.2.2","3.2.2-rc.1","3.2.3","3.2.3-rc.1","3.2.3-rc.2","3.2.4","3.2.4-rc.1","3.2.5","3.2.6","3.2.6-rc.1","3.2.0-a.1","3.2.0-a.2","3.2.0-a.3","3.2.0-a.4","3.2.0-b.1","3.2.0-b.2","3.2.0-rc.1","3.2.0-rc.2","3.2.0-rc.3","3.3.0","3.3.0-a.1","3.3.0-a.2","3.3.0-a.3","3.3.0-a.4","3.3.0-b.1","3.3.0-b.2","3.3.0-rc.1","3.3.0-rc.2","3.3.0-rc.3","3.3.1","3.3.1-rc.1","3.3.2","3.3.3","3.3.3-rc.1","3.3.3-rc.2","3.3.4","3.3.4-rc.1","3.3.5","3.3.5-rc.1","3.3.5-rc.2","3.3.6","3.3.6-rc.1","3.3.7","3.3.7-rc.1","3.4.0","3.4.0-a.1","3.4.0-a.2","3.4.0-a.3","3.4.0-a.4","3.4.0-b.1","3.4.0-b.2","3.4.0-b.3","3.4.0-rc.1","3.4.0-rc.2","3.4.0-rc.3","3.4.1","3.4.1-rc.1","3.4.2","3.4.2-rc.1","3.4.3","3.4.3-rc.1","3.4.4","3.4.4-rc.1","3.4.5","3.4.5-rc.1","3.4.6","3.4.6-rc.1","3.4.7","3.4.7-rc.1","3.4.8","3.4.8-rc.1","3.4.9","3.4.9-rc.1","3.4.10","3.4.10-rc.1","3.5.0","3.5.0-a.1","3.5.0-a.2","3.5.0-a.3","3.5.0-a.4","3.5.0-b.1","3.5.0-b.2","3.5.0-b.3","3.5.0-b.4","3.5.0-rc.1","3.5.0-rc.2","3.5.0-rc.3","3.5.0-rc.4","3.5.1","3.5.1-rc.1","3.5.2","3.5.2-rc.1","3.5.3","3.5.3-rc.1","3.5.4","3.5.4-rc.1","3.5.5","3.5.5-rc.1","3.5.6","3.5.6-rc.1","3.5.7","3.5.7-rc.1","3.5.8","3.5.8-rc.1","3.5.8-rc.2","3.5.9","3.5.10","3.5.10-rc.1","3.6.0","3.6.0-a.1","3.6.0-a.2","3.6.0-a.3","3.6.0-a.4","3.6.0-b.1","3.6.0-b.2","3.6.0-b.3","3.6.0-b.4","3.6.0-rc.1","3.6.0-rc.2","3.6.1","3.6.1-rc.1","3.6.2","3.6.2-rc.1","3.6.2-rc.2","3.6.3","3.6.3-rc.1","3.6.4","3.6.4-rc.1","3.6.5","3.6.5-rc.1","3.6.6","3.6.6-rc.1","3.6.7","3.6.7-rc.1","3.6.7-rc.2","3.6.8","3.6.8-rc.1","3.6.9","3.6.9-rc.1","3.6.10","3.6.10-rc.1","3.6.11","3.6.11-rc.1","3.6.12","3.6.13","3.6.14","3.6.15","3.7.0","3.7.0-a.1","3.7.0-a.2","3.7.0-a.3","3.7.0-a.4","3.7.0-b.1","3.7.0-b.2","3.7.0-b.3","3.7.0-b.4","3.7.0-b.5","3.7.0-rc.1","3.7.1","3.7.1-rc.1","3.7.1-rc.2","3.7.2","3.7.2-rc.1","3.7.3","3.7.3-rc.1","3.7.4","3.7.4-rc.1","3.7.4-rc.2","3.7.5","3.7.5-rc.1","3.7.6","3.7.6-rc.1","3.7.7","3.7.7-rc.1","3.7.8","3.7.8-rc.1","3.7.9","3.7.10","3.7.11","3.7.12","3.7.13","3.7.14","3.7.15","3.7.16","3.7.17","3.8.0","3.8.0-a.1","3.8.0-a.2","3.8.0-a.3","3.8.0-a.4","3.8.0-b.1","3.8.0-b.2","3.8.0-b.3","3.8.0-b.4","3.8.0-rc.1","3.8.1","3.8.1-rc.1","3.8.2","3.8.2-rc.1","3.8.2-rc.2","3.8.3","3.8.3-rc.1","3.8.4","3.8.4-rc.1","3.8.5","3.8.6","3.8.6-rc.1","3.8.7","3.8.7-rc.1","3.8.8","3.8.8-rc.1","3.8.9","3.8.10","3.8.11","3.8.12","3.8.13","3.8.14","3.8.15","3.8.16","3.8.17","3.8.18","3.9.0","3.9.0-a.1","3.9.0-a.2","3.9.0-a.3","3.9.0-a.4","3.9.0-a.5","3.9.0-a.6","3.9.0-b.1","3.9.0-b.2","3.9.0-b.3","3.9.0-b.4","3.9.0-b.5","3.9.0-rc.1","3.9.0-rc.2","3.9.1","3.9.1-rc.1","3.9.2","3.9.2-rc.1","3.9.3","3.9.4","3.9.5","3.9.6","3.9.7","3.9.8","3.9.9","3.9.10","3.9.11","3.9.12","3.9.13","3.9.14","3.9.15","3.9.16","3.9.17","3.9.18","3.10.0","3.10.0-a.1","3.10.0-a.2","3.10.0-a.3","3.10.0-a.4","3.10.0-a.5","3.10.0-a.6","3.10.0-a.7","3.10.0-b.1","3.10.0-b.2","3.10.0-b.3","3.10.0-b.4","3.10.0-rc.1","3.10.0-rc.2","3.10.1","3.10.2","3.10.3","3.10.4","3.10.5","3.10.6","3.10.7","3.10.8","3.10.9","3.10.10","3.10.11","3.10.12","3.10.13","3.11.0","3.11.0-a.1","3.11.0-a.2","3.11.0-a.3","3.11.0-a.4","3.11.0-a.5","3.11.0-a.6","3.11.0-a.7","3.11.0-b.1","3.11.0-b.2","3.11.0-b.3","3.11.0-b.4","3.11.0-b.5","3.11.0-rc.1","3.11.0-rc.2","3.11.1","3.11.2","3.11.3","3.11.4","3.11.5","3.12.0-a.1","3.12.0-a.2","3.12.0-a.3","3.12.0-a.4","3.12.0-a.5","3.12.0-a.6","3.12.0-a.7","3.12.0-b.1","3.12.0-b.2","3.12.0-b.3","3.12.0-b.4","3.12.0-rc.1","3.12.0-rc.2"]}
[TRACE 17:58:03] starbase_utils::fs:install  Creating directory  dir="/Users/SOME_USER/.proto/tools/python"
[TRACE 17:58:03] starbase_utils::fs_lock:install  Locking file exclusively  file="/Users/SOME_USER/.proto/tools/python/remote-versions.json"
[TRACE 17:58:03] starbase_utils::fs_lock:install  Writing file  file="/Users/SOME_USER/.proto/tools/python/remote-versions.json"
[TRACE 17:58:03] starbase_utils::fs_lock:install  Unlocking file exclusively  file="/Users/SOME_USER/.proto/tools/python/remote-versions.json"
[DEBUG 17:58:03] proto_core::tool:install  Resolved to 3.8.18  tool="python" version="3.8.18"
[DEBUG 17:58:03] proto_core::tool:install  Checking if tool is installed  tool="python" install_dir="/Users/SOME_USER/.proto/tools/python/3.8.18"
[DEBUG 17:58:03] proto_core::tool:install  Tool has not been installed  tool="python"
[DEBUG 17:58:03] proto::commands::install:install  Installing Python with version ~3.8
⠁ Installing Python 3.8.18                                                                                         [TRACE 17:58:03] starbase_utils::fs:install  Creating directory  dir="/Users/SOME_USER/.proto/tools/python/3.8.18"
[TRACE 17:58:03] starbase_utils::fs_lock:install  Locking directory  dir="/Users/SOME_USER/.proto/tools/python/3.8.18" pid=15055
[TRACE 17:58:03] starbase_utils::fs:install  Writing file  file="/Users/SOME_USER/.proto/tools/python/3.8.18/.lock"
[DEBUG 17:58:03] proto_core::tool:install  Installing tool natively  tool="python"
[TRACE 17:58:03] warpgate::plugin:install  Calling plugin function native_install  plugin="python" input={"context":{"env_vars":{},"tool_dir":{"path":"/home/.proto/tools/python/3.8.18","virtual_prefix":"/home","real_prefix":"/Users/SOME_USER"},"version":"3.8.18"}}
[TRACE 17:58:03] proto_wasm::exec_command:install  Executing command from plugin  command="which" args=["python-build"] env_vars={}
[TRACE 17:58:03] proto_wasm::exec_command:install  Executed command from plugin  command="which" exit_code=0 stderr_len=0 stdout_len=40
[TRACE 17:58:03] proto_wasm::log:install  Building with `python-build` instead of downloading a pre-built
[TRACE 17:58:03] proto_wasm::exec_command:install  Executing command from plugin  command="python-build" args=["3.8.18", "/Users/SOME_USER/.proto/tools/python/3.8.18"] env_vars={}
━━━        Installing Python 3.8.18                                                                                python-build: definition not found: 3.8.18
[TRACE 17:58:04] proto_wasm::exec_command:install  Executed command from plugin  command="python-build" exit_code=2 stderr_len=0 stdout_len=0
[TRACE 17:58:04] warpgate::plugin:install  Called plugin function native_install  plugin="python" output={"installed":false,"skip_install":false}
[TRACE 17:58:04] starbase_utils::fs_lock:install  Unlocking directory  dir="/Users/SOME_USER/.proto/tools/python/3.8.18"
[TRACE 17:58:04] starbase_utils::fs:install  Removing file  file="/Users/SOME_USER/.proto/tools/python/3.8.18/.lock"
[DEBUG 17:58:04] proto_core::tool:install  Cleaning up temporary files and downloads  tool="python"
[ INFO 17:58:04] proto::commands::install:install  Python has been installed to ~/.proto/tools/python/3.8.18!
[TRACE 17:58:04] starbase::app  Running shutdown phase with 1 systems
```

**Comment by milesj** [comment]
Ok it looks like it did error.

```
━━━        Installing Python 3.8.18                                                                                python-build: definition not found: 3.8.18
[TRACE 17:58:04] proto_wasm::exec_command:install  Executed command from plugin  command="python-build" exit_code=2 stderr_len=0 stdout_len=0
```

I'll looking into bubbling up the error.
